js/recipe/recipe.js

```diff
@@ -0,0 +1,30 @@
+import recipe_data from "../recipe_data.js"
+import { metadata }  from "./metadata.js"
+import { image } from "./image.js"
+import { introduction } from "./introduction.js"
+import { displayIngredients } from "./ingredients.js"
+import { displayMethodSteps } from "./method_steps.js"
+
+export function recipe() {
+  let params = new URLSearchParams(window.location.search);
+  let recipe_id = parseInt(params.get("id"));
+  let recipe = recipe_data.find(recipe => recipe.id == recipe_id);
+
+  return recipe;
+}
+
+let current_recipe = recipe();
+current_recipe.published = true; // recipe is published by default
+
+// TODO: populate these sections:
+// Admin Panel (previous, next, publish, unpublish, edit)
+
+function displayRecipe(recipe) {
+  metadata(recipe);
+  image(recipe);
```

Comment:
If `image` were a component, this could be `render(document.querySelector("article.post.single"), image(recipe))` or similar.

Reply:
Yes! This is what it looks like now actually:
```
render(recipeParentElement, image(recipe));
```

As I have extracted `recipeParentElement` into a constant file.